src/tests/zf_unit/zfattrs.c

```diff
@@ -161,19 +173,94 @@ static void test(struct zf_stack* stack, struct zf_attr* attr,
 
 int main(void)
 {
+  char log_template_1[]="/tmp/logdir1XXXXXX";
+  char log_template_2[]="/tmp/logdir2XXXXXX";
+
+  tmp_log_dir = mkdtemp(log_template_1);
+  if(  tmp_log_dir == NULL )
+    return errno;
+
+  char *non_existent_dir = mkdtemp(log_template_2);
+  if( non_existent_dir == NULL )
+    return errno; 
+
+  int rc = rmdir(non_existent_dir);
+  if( rc < 0 )
+    return rc;
```

Comment:
Why is this creating a tempdir and then immediately deleting it (non_existent_dir)?

Reply:
Because I need to test a non-existent path in:

https://github.com/Xilinx-CNS/tcpdirect/pull/8/files/43dbf5afc8292db609ea74d060aade215e096616#diff-a18c5e88dd663f03845ff5d519eaf2becfed1d9a4a78d05760bee14f250948b1R209

Is there another way you'd prefer ensuring the tested path does not actually exist?

Reply:
I think it would be a little cleaner to avoid doing IO if we can avoid it.

Could we instead generate a temporary file name and stat it to see it doesn't exist?

Reply:
TTBOMK there is not a standard C function that returns only the name of a non-existent file without creating it. I didn't think it worth introducing the complexity of adding a function which generates UID when it's only used once, and for this specific test. Moreover, after writing the UID generator function we'd perhaps start discussing about the probability of collision for that UID. Delegating the responsibility to libc seemed the safest choice. 

I understand that if we had the name only we would avoid a couple of write operation in the disc which will be part of the test anyway as we're writing long logs with 100's of lines (as test process) to the disc. 

Having said that, let me know if you'd be OK with this exception or rather want me to write the UID function; but would appreciate further clarification on why that'd make it cleaner.

Reply:
I don't think you need a C function or uid generator.  "/path/to/nonexistent/file" is very unlikely to exist and you can stat() it as Krish suggested to be sure.

Reply:
Yes, but pedantically speaking it is less safe than mkdtemp and I don't see the problem in terms of the I/O cost. Anyway, as that's the consensus, I will change it to '/path/to/nonexistent/file"